component/metrics/scrape/types.go

```diff
@@ -0,0 +1,286 @@
+	OAuth2ClientSecret                string            `hcl:"oauth2_client_secret,optional"`
+	OAuth2ClientSecretFile            string            `hcl:"oauth2_client_secret_file,optional"`
+	OAuth2Scopes                      []string          `hcl:"oauth2_scopes,optional"`
+	OAuth2TokenURL                    string            `hcl:"oauth2_token_url,optional"`
+	OAuth2EndpointParams              map[string]string `hcl:"oauth2_endpoint_params,optional"`
+	OAuth2ProxyURL                    string            `hcl:"oauth2_proxy_url,optional"`
+	OAuth2TLSConfigCAFile             string            `hcl:"oauth2_tls_config_ca_file,optional"`
+	OAuth2TLSConfigCertFile           string            `hcl:"oauth2_tls_config_cert_file,optional"`
+	OAuth2TLSConfigKeyFile            string            `hcl:"oauth2_tls_config_key_file,optional"`
+	OAuth2TLSConfigServerName         string            `hcl:"oauth2_tls_config_server_name,optional"`
+	OAuth2TLSConfigInsecureSkipVerify bool              `hcl:"oauth2_tls_config_insecure_skip_verify,optional"`
+
+	BearerToken     string `hcl:"bearer_token,optional"`
+	BearerTokenFile string `hcl:"bearer_token_file,optional"`
+	ProxyURL        string `hcl:"proxy_url,optional"`
+
+	TLSConfigCAFile             string `hcl:"tls_config_ca_file,optional"`
+	TLSConfigCertFile           string `hcl:"tls_config_cert_file,optional"`
+	TLSConfigKeyFile            string `hcl:"tls_config_key_file,optional"`
+	TLSConfigServerName         string `hcl:"tls_config_server_name,optional"`
+	TLSConfigInsecureSkipVerify bool   `hcl:"tls_config_insecure_skip_verify,optional"`
+
+	FollowRedirects bool `hcl:"follow_redirects,optional"`
+	EnableHTTP2     bool `hcl:"enable_http_2,optional"`
+}
+
+// DefaultConfig is the set of default options applied before decoding a given
+// scrape_config block.
+var DefaultConfig = Config{
+	MetricsPath:     "/metrics",
+	Scheme:          "http",
+	HonorLabels:     false,
+	HonorTimestamps: true,
+	FollowRedirects: true,                             // From common_config.DefaultHTTPClientConfig
+	ScrapeInterval:  model.Duration(1 * time.Minute),  // From config.DefaultGlobalConfig
+	ScrapeTimeout:   model.Duration(10 * time.Second), // From config.DefaultGlobalConfig
+}
+
+// DecodeHCL implements gohcl.Decoder.
+// This method is only called on blocks, not objects.
+func (c *Config) DecodeHCL(body hcl.Body, ctx *hcl.EvalContext) error {
+	*c = DefaultConfig
+
+	type scrapeConfig Config
+	err := gohcl.DecodeBody(body, ctx, (*scrapeConfig)(c))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Helper function to bridge the in-house configuration with the Prometheus
+// scrape_config.
+// As explained in the Config struct, the following fields are purposefully
+// missing out, as they're being implemented by another components.
+// - RelabelConfigs
+// - MetricsRelabelConfigs
+// - ServiceDiscoveryConfigs
+func (c *Component) getPromScrapeConfigs(scs []Config) []*config.ScrapeConfig {
```

Comment:
nit: Implement a method on Config instead to convert into a ScrapeConfig so you don't need an entire Component to convert them.